### PR TITLE
Checkpoint8 private wikis

### PR DESCRIPTION
**Pull request by oatesmi**
Private wikis dont show up in index for standard users.
Private wikis become not private when downgrading user.
Can not see private wikis unless `record.user == user` or admin.

### CODE REVIEW

File: app/controllers/charges_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class ChargesController < ApplicationController
+
```
Comment (essn) [inline app/controllers/charges_controller.rb]: No space between class definition and callbacks.

File: app/controllers/charges_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  before_action :authenticate_user!
+
+  def new
+  end
+
+  def create
+    @amount = 1500
```
Comment (essn) [inline app/controllers/charges_controller.rb]: This doesn't need to be an instance variable, a regular variable will work just fine.  We only use instance variables in controllers when we want to pass a value to the view.

File: app/controllers/charges_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    # this rescue block catches and displays those errors
+    rescue Stripe::CardError => e
+      flash[:alert] = e.message
+      redirect_to new_charge_path
+  end
+
+  def downgrade
+    current_user.wikis.each do |wiki|
+      wiki.update(:private => false)
+    end
```
Comment (essn) [inline app/controllers/charges_controller.rb]: This logic should be in a method on the User class.  We want to be able to call something like `current_user.publicize_wikis`.

File: app/controllers/charges_controller.rb (hunk 1 above)
Comment (oatesmi) [reply app/controllers/charges_controller.rb]: Removed space.

File: app/controllers/charges_controller.rb (hunk 2 above)
Comment (oatesmi) [reply app/controllers/charges_controller.rb]: Removed instance variable

File: app/controllers/charges_controller.rb (hunk 3 above)
Comment (oatesmi) [reply app/controllers/charges_controller.rb]: Moved this logic in revision for checkpoint7.  Had the same comment from another grader.  Will be reflected on merge to master.

**Pull request merged**